ducktape/tests/runner.py

```diff
@@ -118,11 +123,22 @@ def __init__(self, cluster, session_context, session_logger, tests, deflake_num,
         # This immutable dict tracks test_id -> test_context
         self._test_context = persistence.make_dict(**{t.test_id: t for t in tests})
         self._test_cluster = {}  # Track subcluster assigned to a particular TestKey
-        self._client_procs = {}  # track client processes running tests
+        # track client processes running tests
+        self._client_procs: dict[str, multiprocessing.Process] = {}
```

Comment:
nit: should be typing dict
```suggestion
        self._client_procs: Dict[str, multiprocessing.Process] = {}
```

Reply:
technically both would work, but I'd also prefer typing.Dict

Reply:
Yeah, we only work with python >= 3.9, so we use the lower case variants for collections. I'll switch to the python >3.6 style and make these capitalized.